beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -80,13 +80,24 @@ func (r *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 		return false
 	}
 
-	parentState, err := r.db.State(ctx, bytesutil.ToBytes32(blk.Block.ParentRoot))
-	if err != nil {
-		log.WithError(err).WithField("blockSlot", blk.Block.Slot).Warn("Could not get parent state")
+	// Handle block when the parent is unknown.
+	if !r.db.HasBlock(ctx, bytesutil.ToBytes32(blk.Block.ParentRoot)) {
```

Comment:
this is needed to be put back, if validation fails the block isn't gossiped onwards to other peers.